### PR TITLE
Feature/kc t 12/add support for multivalue

### PR DESCRIPTION
**Pull request by Lacritz**

### CODE REVIEW

File: processor-kotlin/src/main/kotlin/io/datalbry/precise/processor/kotlin/visitor/FindTypesVisitor.kt [hunk 1]
```diff
@@ -10,14 +11,19 @@ import io.datalbry.precise.api.schema.type.DocumentType
...
  * @author timo gruen - 2021-03-11
  */
-class FindTypesVisitor(private val logger: KSPLogger): KSEmptyVisitor<Unit, Set<Type>>() {
-
+class FindTypesVisitor(
+    private val logger: KSPLogger,
+    private val resolver: Resolver
+)
+    : KSEmptyVisitor<Unit, Set<Type>>()
+{
```
Comment (justjanne) [inline processor-kotlin/src/main/kotlin/io/datalbry/precise/processor/kotlin/visitor/FindTypesVisitor.kt]: This doesn't match the formatting of any other piece of code we've got.

```suggestion
): KSEmptyVisitor<Unit, Set<Type>>() {
```

File: processor-kotlin/src/main/kotlin/io/datalbry/precise/processor/kotlin/visitor/FindTypesVisitor.kt [hunk 2]
```diff
@@ -65,3 +97,5 @@ class FindTypesVisitor(private val logger: KSPLogger): KSEmptyVisitor<Unit, Set<
         return name.toLowerCase()
     }
 }
+
+
```
Comment (justjanne) [inline processor-kotlin/src/main/kotlin/io/datalbry/precise/processor/kotlin/visitor/FindTypesVisitor.kt]: Usually we'd only want one LF at the end of a file

```suggestion

```

File: api/src/main/kotlin/io/datalbry/precise/api/schema/field/BasicFieldTypes.kt [hunk 3]
```diff
@@ -36,7 +36,7 @@ enum class BasicFieldTypes(val id: String) {
     /**
      * sequence of 8-bit unsigned bytes
      */
-    BYTES("bytes"),
+    BYTE("byte"),
```
Comment (justjanne) [inline api/src/main/kotlin/io/datalbry/precise/api/schema/field/BasicFieldTypes.kt]: Is this now a sequence of bytes, or a byte?

**Review by justjanne — CHANGES_REQUESTED** [review]
Only minor formatting issues and a single request for clarification, otherwise looks fine

File: api/src/main/kotlin/io/datalbry/precise/api/schema/field/BasicFieldTypes.kt (hunk 3 above)
Comment (Lacritz) [reply api/src/main/kotlin/io/datalbry/precise/api/schema/field/BasicFieldTypes.kt]: Just one byte - the type may be multi valued which would result into a byte sequence. (Basically ByteArray...) 

Going to change the javadoc

File: processor-kotlin/src/main/kotlin/io/datalbry/precise/processor/kotlin/visitor/FindTypesVisitor.kt (hunk 1 above)
Comment (Lacritz) [reply processor-kotlin/src/main/kotlin/io/datalbry/precise/processor/kotlin/visitor/FindTypesVisitor.kt]: Personally I'm using that style of formatting for basically all classes with more than 1 constructor parameter) 
But we can go for the one liner if you prefer

Comment (justjanne) [reply processor-kotlin/src/main/kotlin/io/datalbry/precise/processor/kotlin/visitor/FindTypesVisitor.kt]: It's just a bit confusing, especially with the { on a new line *only* in this case

**Pull request merged**